### ISSUE TITLE
Discussion: Move test suite to Vader

### ISSUE THREAD

**Issue by duggiefresh** [elixir-lang/vim-elixir#104]
tl;dr I am somewhat familiar with VimL and would love to help out on this project, however, it seems that the testing setup is specific to the a contributor's machine.

---

Please correct me if I am wrong: I'm running terminal Vim on iTerm and believe that the testing defaults for this project are a bit of a hinderance. Without too much knowledge of `vimrunner`, it seems that this piece in the [`spec_helper`](https://github.com/elixir-lang/vim-elixir/blob/master/spec/spec_helper.rb#L96) will require the GUI version of Vim. Moreover, it seems that folks will have to compile a version of Vim with [these specific requirements as dictated by `vimrunner` or setup MacVim](https://github.com/AndrewRadev/vimrunner#requirements).

Furthermore, `vimrunner` and `rspec` assumes that a contributor knows some Ruby. Maybe `vim-elixir` could use a VimL friendly testing framework, i.e., (vim-vader)[https://github.com/junegunn/vader.vim]?

Ultimately, if `vimrunner` and `rspec` are going to remain throughout this project, could someone with the domain knowledge add a brief testing setup summary into the `README`. Thanks so much! :smile:

**Comment by rgrinberg** [comment]
+1 for vader as well because you can incorporate vader tests into neovim's CI.

**Comment by whatyouhide** [comment]
:+1: I agree on Vader or, at least, adding some testing setup informations to the README. I personally used Vader myself and I've had only positive experiences.

If this is going somewhere, I'd be happy to try and move the testing to Vader :).

**Comment by kassio** [comment]
Hello guys! I guess that make sense we use Vader. Could someone with Vader experience open a pull request moving the tests to Vader?

**Comment by whatyouhide** [comment]
@kassio It would be a pleasure for me, I'll work on it as soon as I can. :)

**Comment by duggiefresh** [comment]
Sweet! I may have some time in the future and can contribute as well. We can most likely introduce Vader tests piece by piece, and once we have great test coverage, remove rspec and vimrunner all together. Sounds exciting.

**Comment by LnL7** [comment]
That looks amazing! The rspec matchers I wrote always felt like a big hack, but there didn't seem to be a better to do it.

**Comment by kassio** [comment]
Hey guys, any updates on this? May I close this and wait for the pull request?

**Comment by LnL7** [comment]
@kassio I was waiting people's feedback, I can port everything if it looks ok.
I'm still not a fan of duplicating the code examples, but I guess it's fine.

I recently created another vim plugin that uses vader for tests https://github.com/LnL7/vim-nix, that's another example of some syntax and indentation tests.

**Comment by kassio** [comment]
@LnL7 I don't have any problems in switch to vader. @carlosgaldino wdyt?

**Comment by kassio** [comment]
People sorry about the long delay on this one. If anyone could open a Pull Request migrating the tests to vader would be great. I'll close this issue for now hoping to see a pull request soon! :smile: Thank you people!
